File: go/general/1431. 拥有最多糖果的孩子.go
package main

func kidsWithCandies(candies []int, extraCandies int) []bool {
	max_value := max(candies)
	var res []bool
	for _, v := range candies {
		if v+extraCandies >= max_value {
			res = append(res, true)
		} else {
			res = append(res, false)
		}
	}
	return res
}

func max(s []int) int {
	_max := -1
	for _, v := range s {
		if _max < v {
			_max = v
		}
	}
	return _max
}

File: python/general/1431. 拥有最多糖果的孩子.py

from typing import List

class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        res = []
        i = 0
        while i< len(candies):
            candies[i] += extraCandies
            if candies[i] >= max(candies):
                res.append(True)
            else:
                res.append(False)
            candies[i] -= extraCandies
            i+=1
        return res


class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        max_value = max(candies)
        i = 0
        res = []
        while i < len(candies):
            if max_value - candies[i] <= extraCandies:
                res.append(True)
            else:
                res.append(False)
            i+=1
        return res

class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        max_value = max(candies)
        res = [max_value-candies[i]<=extraCandies for i in range(len(candies))]

        return res